src/lib_ccx/ccx_common_common.c

```diff
@@ -57,14 +57,13 @@ void millis_to_time(LLONG milli, unsigned *hours, unsigned *minutes,
 	milli = (milli - *minutes) / 60; // Remainder, in hours
 	*hours = (int)milli;
 }
+
 /* Frees the given pointer */
-void freep(void *arg)
+void freep(void **arg)
 {
-	void **ptr = (void **)arg;
-	if (*ptr)
-		free(*ptr);
-	*ptr = NULL;
-
+	if (*arg)
+		free(*arg);
```

Comment:
Several things on this change:
You're changing freep(), but I don't see you changing any of its callers to now pass a pointer to pointer.
You're not checking if arg itself is NULL, so *arg would coredump by doing this

void *var = NULL
freep (var); // which would happen unless you change the caller
or
freep(&var); // Which would be the correct way now since you changed the signature of freep()

Also, free(NULL) is OK (as in free() itself just returns if passed NULL). So no need to check if (*arg)

Reply:
The callers are already passing pointers to pointers so it should be fine. I just changed the type to reflect that.

> You're not checking if arg itself is NULL, so *arg would coredump by doing this

Fixed, but was already the case previously